src/main/java/net/barrage/school/java/ecatalog/app/services/MerchantServiceImpl.java

```diff
@@ -17,6 +19,17 @@
 public class MerchantServiceImpl implements MerchantService{
     @Autowired
     private MerchantRepository merchantRepository;
+    private final MeterRegistry meterRegistry;
+
+    @Autowired
+    public MerchantServiceImpl(MeterRegistry meterRegistry) {
+        this.meterRegistry = meterRegistry;
+
+        // Register a Gauge for the number of products
+        Gauge.builder("ecatalog.merchants.count", this, MerchantService::countMerchants)
```

Comment:
(just a comment)

In general it's OK, but u have to understand that `countMerchants` will be called every 5 seconds because of scrapper interval
![image](https://github.com/perkeje/barrage-school-ecatalog/assets/136442850/085b5d8c-86f8-4824-af27-4ff6fed11fc1)